Lib/idlelib/debugger.py

```diff
@@ -10,12 +10,20 @@
 
 
 class Idb(bdb.Bdb):
+    """Idle debugger, based on the bdb debugger."""
 
     def __init__(self, gui):
         self.gui = gui  # An instance of Debugger or proxy of remote.
-        bdb.Bdb.__init__(self)
+        self.botframe = None
+        super(Idb, self).__init__()
```

Comment:
You can use the form `super().__init__()`.

Reply:
I forget we're writing Python 3-only code :-)

Reply:
Enhancements to IDLE are now just backported to 3.7, so you can use any language enhancements through that release, including f-strings and even dataclasses, although the latter hasn't been used yet.  :-)

It's an interesting dichotomy - 'cutting edge' code combined with (as you found) 17 year old logic.